### PR TITLE
refactor(delete): clean up error handling and validation

### PR DESCRIPTION
**Pull request by xenonnn4w**
## Changes

- add wrapDeleteError to unify delete-specific error handling
- move delete error types to cmd/errors.go
- implement Unwrap() for errors.Is/As compatibility
- use NewErrNotInitializedFromPath for delete command
- introduce shared ErrNameAndPathConflict (reuse across delete and describe)
- simplify delete command flow
- update and add tests for validation and edge cases

## Tests

- Updated:
  - `TestDelete_NameAndPathExclusivity` to assert `ErrNameAndPathConflict`
- Added:
  - `TestDelete_NoFunctionAtPath`

/kind cleanup

**Release Note**
```release-note
Refactored delete command error handling for consistency and improved validation messages.

### CODE REVIEW

**Comment by gauron99** [comment]
thanks for the contribution! This is a nice change, however to the underlying issue -- this is only part of the solution I had in mind. It wasnt exactly a well-written issue to be picked up, hence I assigned myself, sorry :smile: 
Therefore I would like to keep the issue open. If you want we can split the issue in 2 -> error handling (what you did) and code improvements (what I need to think through how to do)

File: cmd/errors.go [hunk 1]
```diff
@@ -376,6 +422,85 @@ func (e *ErrNotInitialized) Unwrap() error {
...
+// -------------------------------------------------------------------------- //
+
+// ErrDeleteNamespaceRequired wraps core library errors when namespace is missing for delete
+type ErrDeleteNamespaceRequired struct {
+	Err error
+}
+
+// NewErrDeleteNamespaceRequired creates a new ErrDeleteNamespaceRequired
+func NewErrDeleteNamespaceRequired(err error) ErrDeleteNamespaceRequired {
+	return ErrDeleteNamespaceRequired{Err: err}
```
Comment (gauron99) [inline cmd/errors.go]: please use a pointer here for consistency with the other cmd errors (check `NewErrRegistryRequired` or similar)

File: cmd/errors.go [hunk 2]
```diff
@@ -376,6 +422,85 @@ func (e *ErrNotInitialized) Unwrap() error {
...
+}
+
+// Error implements the error interface with CLI-specific help text
+func (e ErrDeleteNamespaceRequired) Error() string {
```
Comment (gauron99) [inline cmd/errors.go]: same as above

File: cmd/errors.go [hunk 3]
```diff
@@ -376,6 +422,85 @@ func (e *ErrNotInitialized) Unwrap() error {
...
+  func delete myfunction --namespace apps    Delete from a specific namespace
+
+Or delete by path if the function has a recorded namespace:
+  func delete --path /path/to/function
+
+For more options, run 'func delete --help'`, e.Err)
+}
+
+// Unwrap returns the underlying error for errors.Is/As
+func (e ErrDeleteNamespaceRequired) Unwrap() error {
```
Comment (gauron99) [inline cmd/errors.go]: same as above

File: cmd/errors.go [hunk 4]
```diff
@@ -376,6 +422,85 @@ func (e *ErrNotInitialized) Unwrap() error {
...
+// ErrDeleteNameRequired wraps core library errors with CLI-specific context
+// for delete operations that require a function name or path.
+type ErrDeleteNameRequired struct {
+	// Underlying error from the core library (e.g., fn.ErrNameRequired)
+	Err error
+}
+
+// NewErrDeleteNameRequired creates a new ErrDeleteNameRequired wrapping the given error
+func NewErrDeleteNameRequired(err error) ErrDeleteNameRequired {
+	return ErrDeleteNameRequired{Err: err}
```
Comment (gauron99) [inline cmd/errors.go]: please use a pointer here for consistency with the other cmd errors (check NewErrRegistryRequired or similar)

File: cmd/errors.go [hunk 5]
```diff
@@ -376,6 +422,85 @@ func (e *ErrNotInitialized) Unwrap() error {
...
+}
+
+// Error implements the error interface with CLI-specific help text
+func (e ErrDeleteNameRequired) Error() string {
```
Comment (gauron99) [inline cmd/errors.go]: same

File: cmd/errors.go [hunk 6]
```diff
@@ -376,6 +422,85 @@ func (e *ErrNotInitialized) Unwrap() error {
...
+Examples:
+   func delete myfunction                     Delete 'myfunction' from cluster
+   func delete myfunction --namespace prod    Delete from 'prod' namespace
+   func delete --path ./myfunction            Delete function at path
+
+For more options, run 'func delete --help'`, e.Err)
+}
+
+// Unwrap returns the underlying error for errors.Is/As
+func (e ErrDeleteNameRequired) Unwrap() error {
```
Comment (gauron99) [inline cmd/errors.go]: same

**Review by gauron99 — COMMENTED** [review]
just a nit from me

**Comment by xenonnn4w** [comment]
Thank you for the review. I have applied your suggestion regarding uniformity! Also i will do a comment under the issue mentioning what is achieved under this pr so you edit the issue as per time in regard of  "code improvements (what I need to think through how to do)" or i'll look more into it and confirm what my views on code improvements!
